src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlParameter.cs

```diff
@@ -978,9 +1005,15 @@ private void CloneHelper(SqlParameter destination)
             destination._offset = _offset;
             destination._sourceColumn = _sourceColumn;
             destination._sourceVersion = _sourceVersion;
-            destination._sourceColumnNullMapping = _sourceColumnNullMapping;
-            destination._isNullable = _isNullable;
-
+            destination._flags = _flags & (
+                SqlParameterFlags.SourceColumnNullMapping |
+                SqlParameterFlags.IsNull |
+                SqlParameterFlags.IsNullable |
+                SqlParameterFlags.IsSqlParameterSqlType |
+                SqlParameterFlags.CoercedValueIsDataFeed |
+                SqlParameterFlags.CoercedValueIsSqlType |
+                SqlParameterFlags.ForceColumnEncryption
```

Comment:
These two are the SqlParameter properties that describe it and they're part of the SqlParameter object characteristic. So, I don't see any reason why they shouldn't be copied to a cloned object!

Any thoughts @David-Engel

Reply:
@Wraith2 Why are all the elements of the SqlParameterFlags enum not included here? I see that HasReceivedMetadata, IsDerivedParameterTypeName, and HasScale are not included. I think that may be what Davoud is really asking. (Or what is special about this subset of items that only they are included?)

Reply:
I asked the question earlier, https://github.com/dotnet/SqlClient/pull/1064#discussion_r629482228 and the answer about why is that because they weren't copied in the original. I think they should be, or at least the two I identified. Doing so is a behavioural change though and I avoid those in these merge PR's and call out when I do make them.

If we all agree that they should be included I can add them. You need to be aware that it might change behaviour though.

Reply:
Ah, I see. HasScale has comments around it indicating it should be ignored
```
private bool _hasScale; // V1.0 compat, ignore _hasScale
```
and that code goes way back in netfx. So I don't care about that one.

But IsDerivedParameterTypeName was recently added in #1020, so I'm good with correcting that here and changing the behavior, assuming it was an oversight there.

HasReceivedMetadata is also relatively new, added for AE in SQL 2016. Looking at how it is used, I'm inclined to leave it as-is. (But I'm not saying it's correct or not.)

Reply:
@Wraith2  One reason for wrapping these values in `SqlParameter` would be drawing special attention to it at the time of adding a new parameter in the future. I believe adding a new member to SqlParameter like **Clonables** would clarify it and it doesn't have any downsides.

Reply:
Ok, done and a comment left to indicate that those were considered so no-one else has to wonder in future.